lib/full-site-editing/block-templates.php

```diff
@@ -386,41 +386,43 @@ function gutenberg_get_block_template( $id, $template_type = 'wp_template' ) {
  * @param string $post_type     Post type.
  * @return string The original, desired slug.
  */
-function gutenberg_filter_wp_template_unique_post_slug( $slug, $post_ID, $post_status, $post_type ) {
-	if ( 'wp_template' !== $post_type || 'wp_template_part' !== $post_type ) {
+function gutenberg_filter_wp_template_unique_post_slug( $placeholder, $slug, $post_ID, $post_status, $post_type ) {
+	if ( 'wp_template' !== $post_type && 'wp_template_part' !== $post_type ) {
 		return $slug;
 	}
 
 	// Template slugs must be unique within the same theme.
-	$theme = get_the_terms( $post_ID, 'wp_theme' )[0]->slug;
+	$theme = wp_get_theme()->get_stylesheet();
 
 	$check_query_args = array(
-		'post_name'      => $slug,
+		'post_name__in'  => array( $slug ),
 		'post_type'      => $post_type,
 		'posts_per_page' => 1,
+		'no_found_rows'  => true,
 		'post__not_in'   => $post_ID,
```

Comment:
Just noting - this `post__not_in` line is probably not necessary since the new item does not have the expected taxonomy yet.  It doesn't hurt though.

Reply:
Curious, why is it needed? A lot of this code could benefit from some explanatory comments imo. :)